backends/occa/ceed-occa-basis.c

```diff
@@ -48,21 +54,19 @@ static int CeedBasisApply_Occa(CeedBasis basis, CeedTransposeMode tmode,
       pre /= P;
       post *= Q;
     }
-    if (tmode == CEED_NOTRANSPOSE) {
+    if (nadd) {
       v += nqpt;
     } else {
       u += nqpt;
     }
   }
   if (emode & CEED_EVAL_GRAD) {
-    CeedInt P = basis->P1d, Q = basis->Q1d;
     // In CEED_NOTRANSPOSE mode:
     // u is (P^dim x nc), column-major layout (nc = ndof)
     // v is (Q^dim x nc x dim), column-major layout (nc = ndof)
     // In CEED_TRANSPOSE mode, the sizes of u and v are switched.
-    if (tmode == CEED_TRANSPOSE) {
-      P = basis->Q1d, Q = basis->P1d;
-    }
+    const CeedInt P = add?basis->Q1d:basis->P1d;
```

Comment:
"add" just feels like a damned confusing name for "transpose", especially if we decouple add/overwrite from transpose/notranspose.